File: Java/PrimerParcial/TP4/P6/Expression.java
package ar.itba.edu.POO.PrimerParcial.TP4.P6;

public abstract class Expression{

    public abstract boolean evaluate();

    public  Expression not(){
        return new NotExpression(this);
    }

    public  Expression or(Expression exp){
        return new OrExpression(this,exp);
    }

    public Expression and(Expression exp){
        return new AndExpression(this,exp);
    }

}

File: Ruby/TP10/Ej3/expression.rb
module Expression
  def evaluate
    raise 'Not implemented'
  end
  def not
    NotExpression.new(self)
  end
  def or(expression)
    OrExpression.new(self, expression)
  end
  def and(expression)
    AndExpression.new(self, expression)
  end
end
